app/crm/connectivity/subscribe.py

```
@@ -0,0 +1,141 @@
+            raise SubscriptionRefused(
+                f"This account is a '{installation.connector_key}' connector, "
+                f"which has no webhook subscription to turn on."
+            )
+    except WebhookSubscriptionError as e:
+        # The provider's own refusal, passed through: a merchant asking "why"
+        # gets an answer matching that provider's documentation rather than
+        # our paraphrase. Caught around the whole block so a second branch
+        # inherits it — a provider raising its own error type adds that type
+        # here (and by then the class belongs somewhere both can see).
+        logger.error(f"connectivity: webhook subscription refused — {e}")
+        raise SubscriptionRefused(str(e))
```

Ruff (0.16.3)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)